### ISSUE TITLE
How to use this open-source?

### ISSUE THREAD

**Issue by joyfulbean** [CODEJIN/PWGAN_for_HiFiSinger#1]
I really want to use PWGAN for hifi-singer. Since there is no direction in .md file, little confuse how should I prepare dataset. Could you please describe how to compile and dataset in detail? If you show me the structure of directories, it will be very helpful. Thank you.

**Comment by CODEJIN** [comment]
Hi, joyfulbean,

I don't recommend use this repository. Although this repository is public, the used dataset is private. Instead, If you would like to build a similar structure with the open source CSD dataset, please check the following:

https://github.com/CODEJIN/MLPSinger/blob/master/Pattern_Generator.py

Patterns generated from above will not be fully compatible with this repository, but the output after pattern generation will be similar.

Best regards,

Heejo

**Comment by joyfulbean** [comment]
Thank you so much for your response!!! I think I can try with CSD dataset.
One thing I wonder is the only reason you do not recommend this repository
is because the dataset is private or there is other reasons, too?
And I have three more questions.
1. Is this vocoder actually working?
2. Do you have any pretrained vocoder checkpoint that I can just try
whether it works?
3. Can you help me how to compile this open source? ex. command line?



2021년 9월 23일 (목) 오후 11:58, Heejo You ***@***.***>님이 작성:

**Comment by CODEJIN** [comment]
Hi,

1. It worked when I tested it. However, the quality was limited and I do not use it now.
2. There is a vocoder that I personally use, but there is nothing that can be shared because it is trained with private datasets.
3. This is Python code, so there is not compile process.

**Comment by joyfulbean** [comment]
I have one more question about your source code. 
I've read several papers about Singing Voice Synthesis to find out how many mel-frames in a syllable should I set to onset, nucleus, coda.
 
According to "XiaoiceSing: A High-Quality and Integrated Singing Voice Synthesis System", the number of frames in each symbol is set as follow. Each character sets the same number of Mel-frames, the number of frames in a syllable which is divided by the number of characters in a syllable. 

According to "Korean Singing Voice Synthesis Based on Auto-Regressive Boundary Equilibrium GAN", the number of frames in each symbol is set as follow. Onset and conda set 1 mel-frame and  nucleus set rested mel frame. 

However, in your repositories, you decide to set 2 mel-frames for onset and conda and rested Mel-frames for nucleus. I am wonder how you get up to this way and why you think you should set 2 Mel-frames, not 1. If you have any insight about this please share, it will be very helpful.

**Comment by CODEJIN** [comment]
At that time, one reason for approaching onset and coda with two frames was that the sample rate of HifiSinger was higher than the previous model. Since each mel frame has half information, I wanted to reflect this.
However, I tested both in a repository created after this repository, and as a result, I did not see a big difference. I'm currently using three frames which is referred from N-Singer, but I think there's no problem with setting it to the same length.
